luaui/Widgets/Tests/balance/test_grunts_vs_pawns.lua

```diff
@@ -0,0 +1,79 @@
+function setup()
+	Test.clearMap()
+end
+
+function cleanup()
+	Test.clearMap()
+end
+
+function test()
+	local units = {
+		[0] = "armpw",
+		[1] = "corak"
+	}
+	local n = 20
+
+	local midX, midZ = Game.mapSizeX / 2, Game.mapSizeZ / 2
+	local xOffset = 200
+	local zStep = 30
+	local startZ = midZ - zStep * n / 2
+
+	-- make two lines of units facing each other
+	SyncedRun(function(locals)
```

Comment:
`locals` includes all local variables that the function would have access to. this is awkward, but pretty much the best we can do right now.

This function is serialized, sent to synced, deserialized, executed, return is serialized, sent to unsynced, and then return is deserialized.

Without `debug` in synced (https://github.com/beyond-all-reason/spring/issues/867), we can't transmit upvalues; if we could, we could basically write these functions as we normally would any function.